src/components/account/SavedVehicle.jsx

```diff
@@ -1,36 +1,143 @@
+import { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+import axios from "axios";
+import AddVehicleForm from "./AddVehicleForm";
+import EditVehicleForm from "./EditVehicleForm";
+import { useDispatch, useSelector } from "react-redux";
+import {
+  setVehicleData,
+  deleteVehicle,
+} from "../../utils/slice/accountsSlice.js";
+
 const SavedVehicle = () => {
+  const dispatch = useDispatch();
+  const vehicleData = useSelector((state) => state.accounts.vehicleData);
+  const [showAddForm, setShowAddForm] = useState(false);
+  const [showEditForm, setShowEditForm] = useState(false);
+  const [selectedVehicle, setSelectedVehicle] = useState(null);
+  const { id } = useParams();
+
+  useEffect(() => {
+    const fetchVehicleData = async () => {
+      try {
+        const response = await axios.get(
+          `http://localhost:3000/user/${id}/cars`
+        );
+
+        dispatch(setVehicleData(response.data));
+      } catch (error) {
+        console.error("Error fetching vehicle data:", error);
+      }
+    };
+
+    fetchVehicleData();
+  }, [dispatch, id, showAddForm, showEditForm]);
```

Comment:
edit it to fire only on submission, if needed further optimization, I could optimistically update the local state with the new data instead of triggering a re-fetch immediately, but it'll be for future optimization.